### PR TITLE
[Code health] Include what you use cleanup, part 5

### PR DESCRIPTION
**Pull request by marcalff**
Contributes to #2054

## Changes

Please provide a brief description of the changes here.

* continued the include-what-you-use cleanup
* total number of warnings decreased from 37 in branch main, to 2 in this PR.
* CI is now enforcing a pass/fail check, with some tolerance (<= 10 warnings)

Note that the cleanup is not completed, as some optional parts (OTLP GRPC) are not covered yet by the build in CI.
This will be handled in subsequent PRs.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by marcalff** [comment]
@open-telemetry/cpp-contrib-approvers Ready for review.

**Comment by marcalff** [comment]
@ThomsonTan @lalitb @esigo @owent 

Any chance to approve this cleanup, so it goes into release 1.18.0 ?

File: exporters/otlp/test/otlp_file_client_test.cc [hunk 1]
```diff
@@ -1,34 +1,51 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
-#include "opentelemetry/common/key_value_iterable_view.h"
+#include <google/protobuf/message_lite.h>
```
Comment (owent) [inline exporters/otlp/test/otlp_file_client_test.cc:4]: This header should be included between `protobuf_include_prefix.h` and protobuf_include_suffix.h

File: exporters/otlp/src/otlp_file_client.cc [hunk 2]
```diff
@@ -34,6 +35,7 @@
 #endif
 
 #include <limits.h>
```
Comment (owent) [inline exporters/otlp/src/otlp_file_client.cc:37]: Should we move system's headers and STL's headers before otel-cpp's headers?

File: exporters/otlp/test/otlp_http_metric_exporter_test.cc [hunk 3]
```diff
@@ -1,37 +1,52 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
+#include <gmock/gmock.h>
+#include <google/protobuf/message_lite.h>
```
Comment (owent) [inline exporters/otlp/test/otlp_http_metric_exporter_test.cc:5]: This header should be included between `protobuf_include_prefix.h` and `protobuf_include_suffix.h`

**Review by owent — CHANGES_REQUESTED** [review]

File: exporters/otlp/test/otlp_http_metric_exporter_test.cc (hunk 3 above)
Comment (marcalff) [reply exporters/otlp/test/otlp_http_metric_exporter_test.cc]: Thanks,

moved `google/protobuf/message_lite.h` between `protobuf_include_prefix.h` and `protobuf_include_suffix.h`

File: exporters/otlp/src/otlp_file_client.cc (hunk 2 above)
Comment (marcalff) [reply exporters/otlp/src/otlp_file_client.cc:4]: Fixed, reordered all includes:

* First, all system headers
* Second, all opentelemetry-cpp headers
* Third, all headers related to protobuf

File: exporters/otlp/test/otlp_file_client_test.cc (hunk 1 above)
Comment (marcalff) [reply exporters/otlp/test/otlp_file_client_test.cc]: Fixed.

I think `message_lite.h` itself is warning clean, but it does not hurt to put it between `protobuf_include_prefix.h` and `protobuf_include_suffix.h`, together with protobuf related code.

**Comment by marcalff** [comment]
@owent 

Thanks for the review.
All comments fixed, PTAL.

**Review by esigo — APPROVED** [review]
LGTM
thanks for the PR :)

**Review by owent — APPROVED** [review]
LGTM and thanks.

**Pull request merged**